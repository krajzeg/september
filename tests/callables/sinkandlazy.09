# Test for how sink and lazy parameters are handled by callables

execute_and_print := Object()
execute_and_print::"<call>" = |?expr, ?...operations| {
	print("Value starts as:", expr.resolve())
	for (op) in (operations) {
		op.resolve()
		print("Value is now:", expr.resolve())
	}
}

x := 21
print("This should print first 21, then 42 and 45.")
execute_and_print(x, x = x * 2, x = x + 3)

