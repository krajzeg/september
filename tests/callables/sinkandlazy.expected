This should print first 21, then 42 and 45.
Value starts as: 21
Value is now: 42
Value is now: 45
